### DISCUSSION TITLE
[ZK Stack] - How to bridge customer base token from local reth to zksync l2?

### DISCUSSION THREAD

**Discussion by SuccinctPaul** [zkSync-Community-Hub/zksync-developers#626] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.78.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

To use customer native token, first need to specify one. Then need to bridge the base token from local reth to l2. How to do all process?

### Expected Behavior

To use customer native token, first need to specify one. Then need to bridge the base token from local reth to l2. 

### Current Behavior

No idea.

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To bridge a customer base token from a local reth node to ZKsync L2, you'll need to follow these steps:

1. **Deploy on Local Reth Node**: Ensure your base token is deployed on your local reth node. If you're using the local setup, you might already have addresses with ETH which can be used to interact with your contracts. More details can be found in the [local setup guide](https://docs.zksync.io/zk-stack/running-a-zk-chain/locally#base-layer-is-the-local-reth-node).

2. **Specify the Customer Native Token**: Define the token you want to use as the base token in your L2 contracts. This involves setting up the token contract on your local reth node.

3. **Bridge the Token to L2**: Use the bridge functionality to transfer the base token from L1 (local reth node) to L2 (ZKsync). This process involves interacting with the bridge contracts deployed on both layers. You can find more information on how to perform this in the [Bridging Assets documentation](https://docs.zksync.io/build/developer-reference/bridging-assets).

4. **Interact with Your ZK Chain**: Once the tokens are bridged to L2, you can interact with them using your ZK chain's RPC interface as detailed in the [Using your ZK chain RPC guide](https://docs.zksync.io/zk-stack/running-a-zk-chain/locally#using-your-zk-chain-rpc).

These steps should guide you through the process of using a customer native token and bridging it from a local reth node to ZKsync L2. If you encounter specific errors or need further customization, consider consulting the detailed documentation or seeking additional developer support.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply, marked as answer]
@SuccinctPaul were you able to get this figured out? We recently posted this in the community code for how to customize the ZK chain for an ERC20 base token that you can see if that helps https://code.zksync.io/tutorials/custom-zk-chain/customizing-your-chain

**Reply by SuccinctPaul** [reply]
> @SuccinctPaul were you able to get this figured out? We recently posted this in the community code for how to customize the ZK chain for an ERC20 base token that you can see if that helps https://code.zksync.io/tutorials/custom-zk-chain/customizing-your-chain

Thx for you info. It's helpful.
